### DISCUSSION TITLE
Bug with pointer to array (unsafer unsafeness)

### DISCUSSION THREAD

**Discussion by hamarb123** [dotnet/csharplang#6692] [Q&A]
As a part of the unsafer unsafeness in C# 11, it seems that we can now take pointers to objects like so:
```csharp
object o = new();
object* ptr = &o;
```

But it seems like it doesn't work properly for array types:
```csharp
string[] arr = new string[0];
string[]* ptr1 = &arr; //doesn't work
var ptr2 = &arr; //works
```

VS even seems to think this should be possible:
<img width="615" alt="image" src="https://user-images.githubusercontent.com/25316035/201462339-da476840-d47c-4869-9b03-f572745cd273.png">

In case it isn't clear why this is an issue: the reason this would be an issue is you wouldn't be able to cast another pointer to this type, or specify a parameter of this type.

I assume it is probably a bug or oversight that you can't declare it, just thought I'd mention it and I hope it gets a bug fix in the next compiler/sdk patch.

**Comment by huoyaoyuan** [comment]
It looks like a parser bug when reproduced in sharplab: `object[]*` does exist in error message when there's type mismatch, but declaring a variable results in syntax error.
